Packages/com.unity.inputsystem/InputSystem/Editor/Internal/InputSystemProfilerModule.cs

```diff
@@ -0,0 +1,148 @@
+                view.Add(m_AverageLatencyLabel);
+                view.Add(m_MaxLatencyLabel);
+                view.Add(m_EventProcessingTimeLabel);
+                view.Add(m_DeviceCountLabel);
+                view.Add(m_ControlCountLabel);
+                view.Add(m_StateBufferSizeLabel);
+
+                // Populate the label with the current data for the selected frame.
+                ReloadData();
+
+                // Be notified when the selected frame index in the Profiler Window changes, so we can update the label.
+                ProfilerWindow.SelectedFrameIndexChanged += OnSelectedFrameIndexChanged;
+
+                return view;
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                if (disposing)
+                {
+                    // Unsubscribe from the Profiler window event that we previously subscribed to.
+                    ProfilerWindow.SelectedFrameIndexChanged -= OnSelectedFrameIndexChanged;
+                }
+
+                base.Dispose(disposing);
+            }
+
+            void ReloadData()
+            {
+                var selectedFrameIndex = System.Convert.ToInt32(ProfilerWindow.selectedFrameIndex);
+
+                var updateCount = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.UpdateCountName);
+                var eventCount = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.EventCountName);
+                var eventSizeBytes = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.EventSizeName);
+                var averageLatency = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.AverageLatencyName);
+                var maxLatency = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.MaxLatencyName);
+                var eventProcessingTime = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.EventProcessingTimeName);
+                var stateBufferSizeBytes = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.StateBufferSizeBytesName);
+                var deviceCount = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.DeviceCountName);
+                var controlCount = ProfilerDriver.GetFormattedCounterValue(selectedFrameIndex,
+                    InputStatistics.Category.Name, InputStatistics.ControlCountName);
+
+                m_UpdateCountLabel.text = $"{InputStatistics.UpdateCountName}: {updateCount}";
```

Comment:
This is just a basic label list at the moment. Is it possible to pass any extra information via the profiler API? Or extract sample data to e.g. create a distribution within detail window?

Reply:
There is a [ProfilerRecorder](https://docs.unity3d.com/ScriptReference/Unity.Profiling.ProfilerRecorder.html) that you can use to gather samples, which then allows you to do more statistical computations.

Reply:
@ekcoh it seems to be possible to add more complex UITK elements. I think a list can be fine for now, and then we could add a nicer UI using UITK (there's more info [here](https://docs.unity3d.com/Manual/profiler-customizing-details-view.html) , which you probably have seen; and a more more complex UI example in [this package](https://github.cds.internal.unity3d.com/unity/com.unity.adaptiveperformance/blob/be81b29e9031944bfb7b5d1b4b2c21fa4d36aa7b/Editor/Profiler/AdaptivePerformanceDetailsViewController.cs#L36-L36) where this is the result: ![image](https://github.com/user-attachments/assets/534dcb29-e77f-4feb-8e88-4d9e765a918c)